src/xrGame/Weapon.cpp

```diff
@@ -1332,16 +1334,18 @@ void CWeapon::UpdateHUDAddonsVisibility()
     else if (m_eSilencerStatus == ALife::eAddonPermanent)
         HudItemData()->set_bone_visible(wpn_silencer, TRUE, TRUE);
 
+    bool useSoCName = HudItemData()->m_model->LL_BoneID(wpn_grenade_launcher) == BI_NONE;
```

Comment:
1. It's more consistent to use snake_case for the variable name here.
2. Use SOC name only if COP is unavailable **and** SOC is available (both conditions).
Because if we will use SOC name when it's unavailable, that might introduce confusion in the log. COP name is preferred.
```suggestion
    bool use_soc_name{};
    if (HudItemData()->m_model->LL_BoneID(wpn_grenade_launcher) == BI_NONE)
        use_soc_name = HudItemData()->m_model->LL_BoneID(wpn_grenade_launcher_soc) != BI_NONE;
```

And, yeah, I know the suggested code looks too long and ugly maybe... :(